### PR TITLE
#34179 Prohibit execution of mutating statements if connection is read-only

### LINKED ISSUE #34179: I was able to delete a row in a DB that was read only.
### Description

Due to data security issues, I cannot give more information. But, I had the client DB in read only.
I was able without an issue to remove data from the DB with "DELETE FROM..."_

### DBeaver Version

community Version 23.2.0.202309041200

### Operating System

Windows 11

### Database and driver

_No response_

### Steps to reproduce

_No response_

### Additional context

_No response_

### PR DESCRIPTION
**Pull request by E1izabeth**
image

### CODE REVIEW

File: plugins/org.jkiss.dbeaver.model/src/org/jkiss/dbeaver/model/sql/SQLQuery.java [hunk 1]
```diff
@@ -510,6 +515,30 @@ public boolean isDropTableDangerous() {
...
+            return false;
+        }
+        if (statement instanceof Select) {
+            SelectBody selectBody = ((Select) statement).getSelectBody();
+            if (selectBody instanceof PlainSelect plainSelectBody) {
+                return plainSelectBody.isForUpdate() || plainSelectBody.getIntoTables() != null;
+            }
+            return false;
+        } else {
+            return true;
```
Comment (LonwoLonwo) [inline plugins/org.jkiss.dbeaver.model/src/org/jkiss/dbeaver/model/sql/SQLQuery.java:531]: Just to check - is it expected to have "true" on non-select statement in isModifyingSelect method?

Comment (E1izabeth) [reply plugins/org.jkiss.dbeaver.model/src/org/jkiss/dbeaver/model/sql/SQLQuery.java:531]: I think, I misunderstood this method.
P.S. I haven't changed it, just moved and renamed.

**Review by serjiokov — APPROVED** [review]

**Review by LonwoLonwo — APPROVED** [review]

**Comment by uslss** [comment]
verified

File: plugins/org.jkiss.dbeaver.model/src/org/jkiss/dbeaver/model/sql/SQLQuery.java [hunk 2]
```diff
@@ -510,6 +510,21 @@ public boolean isDropTableDangerous() {
...
```
Comment (serge-rider) [inline plugins/org.jkiss.dbeaver.model/src/org/jkiss/dbeaver/model/sql/SQLQuery.java:531]: No sure we can use this approach. There are many non-SELECT queries which are not modifying. E.g. USE, SHOW, WITH, etc. Every dialect may have extra allowed statements.
Let's restrict only INSERT, DELETE, UPDATE, CREATE, DROP, ALTER for now.

**Review by serge-rider — CHANGES_REQUESTED** [review]

**Comment by uslss** [comment]
verified

**Pull request merged**